libs/ledger-live-common/src/load/speculos.ts

```diff
@@ -316,8 +316,11 @@ export function appCandidatesMatches(
           hackBadSemver(appCandidate.firmware),
           searchFirmware
         ))) &&
-    (!search.appVersion ||
-      semver.satisfies(appCandidate.appVersion, search.appVersion))
+    ((!search.appVersion &&
+      !appCandidate.appVersion.includes("prerelease") &&
+      !appCandidate.appVersion.includes("rc")) ||
+      (search.appVersion &&
+        semver.satisfies(appCandidate.appVersion, search.appVersion)))
```

Comment:
I can see the use of a `hackBadSemver` just above, on L316
Should we use the same logic to be consistent?

Reply:
mmh i'm not sure to follow your feedback. I only added 

```
 &&
      !appCandidate.appVersion.includes("prerelease") &&
      !appCandidate.appVersion.includes("rc"))
```
      
from the previous code

your suggestion is not related to the fix

Reply:
You're right nevermind, `hackBadSemver` is only to fix syntax but doesn't filter out anything...